### ISSUE TITLE
Unable to choose Fully preemtible model in version 5.15.65

### ISSUE THREAD

**Issue by Larsinator7** [raspberrypi/linux#5168]
![Screenshot from 2022-09-08 18-00-56](https://user-images.githubusercontent.com/113099306/189171369-b912f3c3-f581-46ec-8ab1-a5a73cfae879.png)
### Describe the bug

I'm trying to get a fully preemtible model to work on a Raspberry Pi using cross compiling. I've encountered an issue on the newest version of the github repo that I'm unable to choose the option of the fully preemtible model in the menuconfig. I'm trying to add a fully preemtible kernel on rpi version 5.15.65 with the corresponding patch. Any reasons why this problem might occur, alternatively what corresponding version and patches will work? I've tried several but I get different issues. 

### Steps to reproduce the behaviour

Downloading the github repo

### Device (s)

Other

### System

Kernel version: 5.15.55-rt48 #2 SMP PREEMPT_RT Mon Sep 5 19:09:07 CEST 2022 x86_64 x86_64 x86_64 GNU/Linux
Done on a host computer running Ubuntu version 22.04

### Additional context

I did the patch of the Kernel on my computer myself if that is relevant.
Edit: Added a pic of the menuconfig

**Comment by Larsinator7** [comment]
PS: Could it be the patch that is troublesome?

**Comment by pelwell** [comment]
It works for me:
```
$ diff 1 2
108,109c108,111
< CONFIG_PREEMPT_VOLUNTARY=y
< # CONFIG_PREEMPT is not set
---
> # CONFIG_PREEMPT_VOLUNTARY is not set
> CONFIG_PREEMPT=y
> CONFIG_PREEMPT_COUNT=y
> CONFIG_PREEMPTION=y
131a134
> CONFIG_PREEMPT_RCU=y
135a139
> CONFIG_TASKS_RCU=y
749,753c753
< CONFIG_INLINE_SPIN_UNLOCK_IRQ=y
< CONFIG_INLINE_READ_UNLOCK=y
< CONFIG_INLINE_READ_UNLOCK_IRQ=y
< CONFIG_INLINE_WRITE_UNLOCK=y
< CONFIG_INLINE_WRITE_UNLOCK_IRQ=y
---
> CONFIG_UNINLINE_SPIN_UNLOCK=y
4150a4151
> # CONFIG_CEC_GPIO is not set
8088a8090
> CONFIG_DEBUG_PREEMPT=y
8166a8169
> # CONFIG_PREEMPT_TRACER is not set
```
(where `1` is the result of `make ARCH=arm bcm2711_defconfig` and `2` is after `make ARCH=arm menuconfig` and changing to `Preemptible Kernel`).

Which sequence of commands did you run?

**Comment by Larsinator7** [comment]
I did:

$ gzip -cd ../patch-5.15.65-rt49.patch.gz | patch -p1 --verbose
$ make O=../kernel-out/ ARCH=arm64 CROSS_COMPILE=/opt/aarch64/bin/aarch64-linux-gnu- bcm2711_defconfig
$ make O=../kernel-out/ ARCH=arm64 CROSS_COMPILE=/opt/aarch64/bin/aarch64-linux-gnu- menuconfig

Patch version is [[patch-5.15.65-rt49.patch.gz](https://mirrors.edge.kernel.org/pub/linux/kernel/projects/rt/5.15/patch-5.15.65-rt49.patch.gz)

and it just didn't show up as an option in the menu

**Comment by pelwell** [comment]
I think the RT patch may have replaced or disabled the `CONFIG_PREEMPT` option. If you've installed the RT patches. wouldn't you want to enable `CONFIG_PREEMPT_RT` instead?

**Comment by Larsinator7** [comment]
![Screenshot from 2022-09-08 14-34-56](https://user-images.githubusercontent.com/113099306/189124208-c8c1e5a2-fec7-4051-b48a-18b5abfe7b67.png)
I went into the config file and tried to enable the CONFIG_PREEMPT_RT option but this doesn't seems to exist in the config file with the current patch. Atleast I can't find it when searching for it

Edit: I tried the same patch as I did on my PC with the linux-rpi-5.15.65 repo and I still have the same problem. Are you sure it's not something wrong in the repo that disables the possibility of the Fully Preemptive (Real-time) model?
